loader/checkpoint.go

```diff
@@ -395,15 +395,23 @@ func (cp *RemoteCheckPoint) GenSQL(filename string, offset int64) string {
 }
 
 // UpdateOffset implements CheckPoint.UpdateOffset.
-func (cp *RemoteCheckPoint) UpdateOffset(filename string, offset int64) {
+func (cp *RemoteCheckPoint) UpdateOffset(filename string, offset int64) error {
 	cp.restoringFiles.Lock()
 	defer cp.restoringFiles.Unlock()
 	db, table, err := getDBAndTableFromFilename(filename)
 	if err != nil {
-		cp.logger.Error("error in checkpoint UpdateOffset", zap.Error(err))
-		return
+		return terror.Annotatef(terror.ErrLoadTaskCheckPointNotMatch.Generate(), "wrong filename=%s", filename)
 	}
-	cp.restoringFiles.pos[db][table][filename][0] = offset
+
+	if _, ok := cp.restoringFiles.pos[db]; ok {
+		if _, ok := cp.restoringFiles.pos[db][table]; ok {
+			if _, ok := cp.restoringFiles.pos[db][table][filename]; ok {
+				cp.restoringFiles.pos[db][table][filename][0] = offset
+				return nil
+			}
+		}
+	}
+	return terror.Annotatef(terror.ErrLoadTaskCheckPointNotMatch.Generate(), "db=%s table=%s not in checkpoint", db, filename)
```

Comment:
terror.ErrLoadTaskChecPointNotMatch.AnnotateDelegate or terror.Annotate(err,...) We should keep the origin err.

Reply:
- -. L414 is no origin error any more, should i use `errors.New("map key not found")` ?